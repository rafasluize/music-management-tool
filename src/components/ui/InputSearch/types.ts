import { IInput } from '../InputField/types';

export interface IInputSearch extends IInput {
  onChange: any;
}
