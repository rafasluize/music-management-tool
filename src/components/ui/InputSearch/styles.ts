import { ErrorMessage } from 'formik';
import styled from 'styled-components';

const Wrapper = styled.div`
  margin-bottom: ${({ theme }) => theme.spacing.km4};

  > div {
    outline: ${({ theme }) => theme.border.width.thinner} solid ${({ theme }) => theme.color.text.neutral.weak};
    display: flex;
    align-items: center;

    > input {
      height: 30px;
      border: none;
      width: calc(100% - 16px);

      &:focus {
        outline: none;
      }
    }
  }

  > label {
    display: block;
    font-size: ${({ theme }) => theme.font.size.m3};
    margin-bottom: ${({ theme }) => theme.spacing.km1};
  }
`;

const ErrorMessageCustom = styled(ErrorMessage)`
  color: ${({ theme }) => theme.color.text.danger.default};
  font-size: ${({ theme }) => theme.font.size.m3};
`;

export { Wrapper, ErrorMessageCustom };
