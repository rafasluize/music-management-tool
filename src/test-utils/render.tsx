import React, { FC, JSXElementConstructor, ReactElement, ReactNode, Suspense } from 'react';
import { ThemeProvider } from 'styled-components';
import { render, RenderOptions } from '@testing-library/react';
import theme from 'styles/theme';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { BrowserRouter } from 'react-router-dom';

const ThemeDS: FC<{ children: ReactNode }> = ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>;

export const wrapperRender = (
  component: ReactElement<Element, string | JSXElementConstructor<Element>>,
  options?: RenderOptions<typeof import('@testing-library/dom/types/queries'), HTMLElement>,
) => render(component, { wrapper: ThemeDS, ...options });

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

const queryRender = (
  component: ReactElement<Element, string | JSXElementConstructor<Element>>,
  defaultValues?: unknown,
  options?: RenderOptions<typeof import('@testing-library/dom/types/queries'), HTMLElement>,
) =>
  render(
    <Suspense fallback={<p>Carregando</p>}>
      <QueryClientProvider client={queryClient}>
        <BrowserRouter>{component}</BrowserRouter>
      </QueryClientProvider>
    </Suspense>,
    { wrapper: ThemeDS, ...options },
  );

export { wrapperRender as render, queryRender };

export default wrapperRender;
