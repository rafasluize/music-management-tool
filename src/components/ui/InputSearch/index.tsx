import React, { FC } from 'react';
import { IInputSearch } from './types';
import * as Styled from './styles';
import { MdSearch } from 'react-icons/md';

const InputSearch: FC<IInputSearch> = ({ label, name, type, onChange }) => {
  return (
    <Styled.Wrapper>
      {label && <label htmlFor={name}>{label}</label>}

      <div>
        <input id={name} type={type} name={name} onChange={onChange} />
        <MdSearch color="#babec9" />
      </div>
    </Styled.Wrapper>
  );
};

export default InputSearch;
