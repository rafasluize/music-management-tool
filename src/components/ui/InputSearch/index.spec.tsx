import React from 'react';
import { render } from 'test-utils/render';
import InputSearch from '.';

describe('input search', () => {
  it('snapshot', () => {
    const { container } = render(<InputSearch label="search" name="search" onChange={() => null} type="text" />);

    // eslint-disable-next-line testing-library/no-node-access
    const wrapper = container.firstElementChild;

    expect(wrapper).toMatchSnapshot();
  });
});
