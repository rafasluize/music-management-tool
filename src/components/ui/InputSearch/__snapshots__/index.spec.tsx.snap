// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input search snapshot 1`] = `
.c0 {
  margin-bottom: 16px;
}

.c0 >div {
  outline: 1px solid #babec9;
  display: flex;
  align-items: center;
}

.c0 >div >input {
  height: 30px;
  border: none;
  width: calc(100% - 16px);
}

.c0 >div >input:focus {
  outline: none;
}

.c0 >label {
  display: block;
  font-size: 0.875rem;
  margin-bottom: 4px;
}

<div
  class="c0"
>
  <label
    for="search"
  >
    search
  </label>
  <div>
    <input
      id="search"
      name="search"
      type="text"
    />
    <svg
      color="#babec9"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(186, 190, 201);"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0 0 16 9.5 6.5 6.5 0 1 0 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
    </svg>
  </div>
</div>
`;
